packages/apptive_grid_form/lib/src/widgets/form_widget/form_content_widgets.dart

```diff
@@ -96,128 +92,94 @@ class FormDataWidget extends StatelessWidget {
-            }
-            final componentWidget =
-                fromModel(component, enabled: properties?.disabled != true);
-            if (componentWidget is EmptyFormWidget) {
-              // UserReference Widget should be invisible in the Form
-              // So returning without any Padding
-              return componentWidget;
-            } else {
-              return IgnorePointer(
-                ignoring: properties?.disabled == true || isSubmitting,
-                child: Padding(
-                  padding: padding,
-                  child: Builder(
-                    builder: (context) {
-                      final customBuilder =
-                          componentBuilder?.call(context, component);
-                      if (customBuilder != null) {
-                        return customBuilder;
-                      } else {
-                        return componentWidget;
-                      }
-                    },
-                  ),
-                ),
-              );
-            }
-          } else {
-            return Padding(
-              padding: padding,
-              child: Align(
-                alignment: buttonAlignment,
-                child: Builder(
-                  builder: (_) {
-                    if (isSubmitting) {
-                      return Column(
-                        mainAxisSize: MainAxisSize.min,
-                        crossAxisAlignment: CrossAxisAlignment.stretch,
-                        children: [
-                          const TextButton(
-                            onPressed: null,
-                            child: Center(
-                              child: CircularProgressIndicator.adaptive(),
-                            ),
-                          ),
-                          if (progress != null)
-                            Padding(
-                              padding: padding,
-                              child: SubmitProgressWidget(progress: progress!),
-                            ),
-                        ],
-                      );
-                    } else if (!hideButton) {
-                      return ElevatedButton(
-                        onPressed: submitForm,
-                        child: Text(
-                          buttonLabel ??
-                              data.properties?.buttonTitle ??
-                              localization.actionSend,
-                        ),
-                      );
-                    } else {
-                      return const SizedBox();
-                    }
-                  },
-                ),
+    final pages = widget.data.properties?.pageIds;
+    if (pages != null && pages.length > 1) {
+      return PageView(
+        controller: _pageController,
+        physics: const NeverScrollableScrollPhysics(),
+        children: pages.map(
+          (pageId) {
+            final key = GlobalKey<FormState>(debugLabel: pageId);
```

Comment:
@ABausG do you mean the keys or the full pages?

Reply:
Yeah the issue is here that you should not create Global Keys inside a build method because that might result in the key that is used to then verify the page might not be the one that is currently attached to the widget tree.

Reply:
So basically the keys should be stored basically as a Map of `pageId:GlobalKey<FormState>` and I guess that can either happen in `initState` or by using `putIfAbsent` on that map to create them when needed. However I guess initializing them directly is potentially cleaner syntactically

Reply:
done